src/puppetlabs/pcp/message.clj

```diff
@@ -121,14 +122,39 @@
 (s/defn make-message :- Message
```

Comment:
`make-message` doesn't return something matching either `Message` or `v1-Message`. It's also practically unused, so why keep both it and `make-versioned-message`?

There's no need to maintain backward compatibility of functions in this package. Only the protocol itself.